pkg/controller/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -271,9 +280,31 @@ func TestReconcileProvisionFail(t *testing.T) {
 
 		// test
 		reconcile(r, req, "unable to create some object")
+
 		checkStatus(t, fakeClient, username, corev1.ConditionFalse, "UnableToProvisionNamespace")
 	})
 
+	t.Run("fail_list_namespace", func(t *testing.T) {
+		r, req, fakeClient := prepareReconcile(t, username, nsTmplSet)
+		fakeClient.MockList = func(ctx context.Context, opts *client.ListOptions, list runtime.Object) error {
+			return errors.New("unable to list namespace")
+		}
+
+		// test
+		reconcile(r, req, "unable to list namespace")
+
+		checkStatus(t, fakeClient, username, corev1.ConditionFalse, "UnableToProvision")
+	})
+
+	t.Run("fail_get_nstmplset", func(t *testing.T) {
+		r, req, fakeClient := prepareReconcile(t, username)
+		fakeClient.MockGet = func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+			return errors.New("unable to get NSTemplate")
+		}
+
+		// test
+		reconcile(r, req, "unable to get NSTemplate")
```

Comment:
Same here.

Reply:
Similarly, this is error (mocked) for get NSTmplSet and also No given NSTmplSet object available.  Open for suggestions here.

Reply:
Yep. Looks good. Sorry for the false alarm.